elastic/kibana
# [Lens] gracefully handle JS Object method names as field names

## Summary

The changed part of the code uses directly the field names as indices of a plain JS object.
This causes an issue internally when a field name called `toString` pass through that code. The code was called with the following:
```
supportedOperationsByField[`toString`]?.add(operation.operationType);
```
but `supportedOperationsByField` is already an object and has the static method `toString`. This causes the throw when calling the `add()` operation into the `toString` method.

The fix reconfigured that part of the code by replacing plain Objects with `Map` that doesn't suffer from this issue.

A test was added before the commit to catch the failure and verify the fix.

fix #148062

### To test

To double-check locally the behavior you can create a new runtime field called `toString` or any other name of methods of a plain JS Object prototype (`constructor`, `valueOf` etc).



### Checklist
